File: .leetcode/674.最长连续递增序列.cpp
/*
 * @lc app=leetcode.cn id=674 lang=cpp
 *
 * [674] 最长连续递增序列
 */

// @lc code=start
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findLengthOfLCIS(vector<int>& nums) {
        if(!nums.size())return 0;
        vector<int>dp(nums.size(),1);
        int ans = 1;
        for(int i = 1;i<nums.size();i++){
            if(nums[i]>nums[i-1]){
                dp[i] = dp[i-1]+1;
                ans = max(ans,dp[i]);
            }
        }
        return ans;
    }
};
// @lc code=end

File: .leetcode/674.最长连续递增序列.java
/*
 * @lc app=leetcode.cn id=674 lang=java
 *
 * [674] 最长连续递增序列
 */

// @lc code=start
class Solution {
    public int findLengthOfLCIS(int[] nums) {
        int ans = 1;
        int temp = 0;
        int pre = nums[0];
        for(int i = 0;i < nums.length;i++){
            if(nums[i] > pre){
                temp++;
                ans = Math.max(ans, temp);
            }else{
                temp = 1;
            }
            pre = nums[i];
        }
        return ans;
    }
}
// @lc code=end

